src/runtime/storageNG/store.ts

```diff
@@ -79,6 +91,35 @@ export class DirectStore<T extends CRDTTypeRecord> extends ActiveStore<T> {
     super(storageKey, exists, type, mode, modelConstructor);
   }
 
+  async idle() {
+    if (this.pendingException) {
+      return Promise.reject(this.pendingException);
+    }
+    if (this.state === DirectStoreState.Idle) {
+      return Promise.resolve();
+    }
+    return new Promise<void>((resolve, reject) => {
+      this.pendingResolves.push(resolve);
+      this.pendingRejects.push(reject);
+    });
+  }
+
+  private setState(state: DirectStoreState) {
+    this.state = state;
+    if (state === DirectStoreState.Idle) {
+      this.notifyIdle();
+    }
+  }
+
+  private notifyIdle() {
+    if (this.pendingException) {
+      this.pendingRejects.forEach(reject => reject(this.pendingException));
```

Comment:
when there's a pendingException we are in an error state and aren't going to recover. There's no explicit need to leave everything in a clean state; but I guess it wouldn't hurt if you think it would make the code easier to grok?